r-package/survlearners/R/utils.R

```diff
@@ -1,92 +1,92 @@
-# For computing baseline hazard in coxph
-base_surv <- function(fit, Y, D, x, lambda){
-  data <- data.frame(t_event=Y, event=D, x)
-  tab <- data.frame(table(data[data$event == 1, "t_event"]))
-  y <- as.numeric(as.character(sort(unique(tab[,1]))))
-  d <- tab[,2]  # number of events at each unique time
-
-  betaHat <- as.vector((fit$glmnet.fit$beta)[,fit$lambda==lambda])
-  h0 <- rep(NA, length(y))
-  for(l in 1:length(y)){
-    h0[l] <- d[l] / sum(exp(x[data$t_event >= y[l], rownames(fit$glmnet.fit$beta)] %*% betaHat))
-  }
-
-  S0 <- exp(-cumsum(h0))
-  outcome <- data.frame(time=y,survival=S0)
-  outcome
-}
-pred_surv <- function(fit, S0, x, times, lambda){
-  link <- predict(fit$glmnet.fit,x,type = "link")[,fit$lambda==lambda]
-  colnames(link) <- NULL
-
-  if(length(times)>1){
-    S0_t <- rep(NA, length(times))
-    for (i in 1:length(times)){
-      S0_t[i] <- S0$survival[S0$time>=times[i]][1]
-    }
-  }else{
-    S0_t <- S0$survival[S0$time>=times][1]
-  }
-
-  surv <- S0_t^exp(link)
-  surv
-}
-pred_surv_preval <- function(fit, S0, times, lambda){
-  link <- fit$fit.preval[,!is.na(colSums(fit$fit.preval))][, fit$lambda[!is.na(colSums(fit$fit.preval))] == lambda]
-  colnames(link) <- NULL
-
-  if(length(times)>1){
-    S0_t <- rep(NA, length(times))
-    for (i in 1:length(times)){
-      S0_t[i] <- S0$survival[S0$time>=times[i]][1]
-    }
-  }else{
-    S0_t <- S0$survival[S0$time>=times][1]
-  }
-
-  surv <- S0_t^exp(link)
-  surv
-}
-
-
-sanitize_x = function(x){
-	# make sure x is a numeric matrix with named columns (for caret)
-	if (!is.matrix(x) | !is.numeric(x) | any(is.na(x))) {
-		stop("x must be a numeric matrix with no missing values")
-	}
-	colnames(x) = paste0("covariate_", 1:ncol(x))
-	return(x)
-}
-
-sanitize_input = function(x,w,y,D) {

```

Comment:
Is  everything in this file being changed or are some things just formatting changes??

From the diff it looks like a lot is only formatting changes you accidentallly copy/pasted?? If so this should not happen here.

It's the same point trying to make before:

this PR contains many bug fixes to the estimator R code

when looking throught the DIFF of what's being fixed you have to trawl/sift through TONS of stuff, this included, and you have to stop and think:

Are bugs in utills.R being fixed?? NO it's just many accidental formatting changes that has nothing to do with the bugs

Reply:
Most of the changes are related to the format. All the codes worked fine before on my local computer for running simulations, but when going through the tests on github, it fails due to things like package specification or formatting issues. I will start a new branch for myself for cleanup code purposes

Reply:
Things like these looks more serious than formatting or package specification:

![Screen Shot 2022-04-09 at 09 26 38](https://user-images.githubusercontent.com/7185264/162582756-f801320f-111c-422e-91b6-623364700e92.png)

which is why I suggested to you to carefully do things step by step